src/content/ui/interactivity/focus.md

```diff
@@ -3,179 +3,226 @@ title: Understanding Flutter's keyboard focus system
-to realize that there really is no such thing as "unfocusing" all nodes. If a
-node is unfocused, then it must pass the focus somewhere else, since there is
-_always_ a primary focus. The node that receives the focus when a node calls
-`unfocus()` is either the nearest `FocusScopeNode`, or a previously focused node
-in that scope, depending upon the `disposition` argument given to `unfocus()`.
-If you would like more control over where the focus goes when you remove it from
-a node, explicitly focus another node instead of calling `unfocus()`, or use the
-focus traversal mechanism to find another node with the `focusInDirection`,
+**Do create `FocusNode` and `FocusScopeNode` objects in a stateful widget.**
+: When you finish using `FocusNode` or `FocusScopeNode`,
+  you must dispose of them.
+  For this reason, only create them inside of a stateful widget's `State` object.
+  You can override the stateful widget's `dispose` method to dispose of the
+  focus objects.
+
+**Do set the `debugLabel` of a focus node widget.**
+: This helps with diagnosing focus issues.
+  The result appears in diagnostic output as `debugDumpFocusTree()`.
+
+**Do call the `requestFocus()` method on a node to request the primary focus.**
+: Call this method from a node that has passed a node
+  it owns to a descendant where you want to the focus.
+
+**Do use `focusNode.requestFocus()`.**
+: You don't need to call `FocusScope.of(context).requestFocus(focusNode)`.
+  The `focusNode.requestFocus()` method works the same with better performance.
+
+**Don't allocate a new `FocusNode` object for each build.**
+: This can cause memory leaks.
+  It might also cause a loss of focus when the widget rebuilds while
+  the node has focus.
+
+**Don't use the same `FocusNode` for multiple widgets.**
+: If you do, the widgets fight over managing the attributes of the node.
+  That might result in an unexpected outcome.
+
+**Don't set attributes on focus nodes without using `Focus.withExternalFocusNode()`**
+: If you want to include attributes on a FocusNode,
+  create a FocusNode outside of the Focus widget.
+  This prevents you from unintentionally overwriting any attributes
+  you set on the [`Focus`][] widget.
+
+  This applies to a `FocusScopeNode` that a [`FocusScope`][] widget
+  manages.
+
+## Tell a node to give up focus
+
+The `FocusNode.unfocus()` API tells a node to "give up the focus".
+While it does remove focus from the node,
+you can't "unfocus" all nodes.
+If you remove focus from a node, it must pass the focus somewhere else,
+since there is _always_ a primary focus. When a node calls `unfocus()`,
+focus can pass to one of two nodes.
+It can pass to either the nearest `FocusScopeNode`
+or a node in the same scope that had focus earlier.
+
+Which node Flutter chooses depends upon the `disposition` argument
+you give to `unfocus()`.
+When calling `unfocus()`, you can set the `disposition` argument.
+It allows two modes for passing focus:
```

Comment:
This text is very verbose and duplicative. Why not make it much more concise, such as:

```suggestion
Flutter chooses a new focus node depending on the `disposition` argument
you pass to `unfocus()`:
```